src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java

```diff
@@ -115,7 +122,39 @@ public List<ExtendedCourthouse> mapFromEntitiesToExtendedCourthouses(List<Courth
 
     @Override
     public List<CourthouseEntity> getAllCourthouses() {
-        return courthouseRepository.findAll();
+
+        UserAccountEntity userAccount = authorisationApi.getCurrentUser();
+
+        if (userAccount != null) {
+            List<SecurityRoleEnum> securityRoles = new ArrayList<>();
+            securityRoles.add(SUPER_ADMIN);
+            securityRoles.add(JUDGE);
+
+            Set<SecurityGroupEntity> securityGroupEntities2 = userAccount.getSecurityGroupEntities();
+            List<Integer> securityRoleIds = securityRoles.stream().map(SecurityRoleEnum::getId).toList();
+            boolean userFound = securityGroupEntities2.stream()
+                .anyMatch(securityGroup -> securityRoleIds.contains(securityGroup.getSecurityRoleEntity().getId()));
+
+
+            if (userFound) {
+                return courthouseRepository.findAll();
+            }
+        }
+
+        return getCourthouseEntities();
+
+    }
+
+    private List<CourthouseEntity> getCourthouseEntities() {
+        List<CourthouseEntity> userCourthouseEntities = new ArrayList<>();
+        List<Integer> courthouseIdsUserHasAccessTo = authorisationApi.getListOfCourthouseIdsUserHasAccessTo();
+
+
+        for (Integer courthouseId : courthouseIdsUserHasAccessTo) {
+            Optional<CourthouseEntity> courthouseEntity = courthouseRepository.findById(courthouseId);
```

Comment:
as mentioned earlier, we should be creating a method in courthouseRepository called findByIdIn